docs/docs/10.6-create-fragment.md

```diff
@@ -0,0 +1,73 @@
+---
+id: create-fragment
+title: Keyed Fragments
+permalink: create-fragment.html
+prev: clone-with-props.html
+next: update.html
+---
+
+In most cases, you can use the `key` prop to specify keys on the elements you're returning from `render`. However, this breaks down in one situation: if you have two sets of children that you need to reorder, there's no way to put a key on each set without adding a wrapper element.
+
+That is, if you have a component such as:
+
+```js
+var Swapper = React.createClass({
+  propTypes: {
+    // `leftChildren` and `rightChildren` can be a string, element, array, etc.
+    leftChildren: React.PropTypes.node,
+    rightChildren: React.PropTypes.node,
+
+    swapped: React.PropTypes.bool
+  }
+  render: function() {
+    var children;
+    if (this.props.swapped) {
+      children = [this.props.rightChildren, this.props.leftChildren];
+    } else {
+      children = [this.props.leftChildren, this.props.rightChildren];
+    }
+    return <div>{children}</div>;
+  }
+});
+```
+
+The children will unmount and remount as you change the `swapped` prop because there aren't any keys marked on the two sets of children.
+
+To solve this problem, you can use `React.addons.createFragment` to give keys to the sets of children.
+
+#### `ReactFragment React.addons.createFragment(object children)`
+
+Instead of creating arrays, we write:
+
+```js
+if (this.props.swapped) {
+  children = React.createFragment({
+    right: this.props.rightChildren,
+    left: this.props.leftChildren
+  });
+} else {
+  children = React.addons.createFragment({
+    left: this.props.leftChildren,
+    right: this.props.rightChildren
+  });
+}
+```
+
+The keys of the passed object (that is, `left` and `right`) are used as keys for the entire set of children, and the order of the object's keys is used to determine the order of the rendered children. With this change, the two sets of children will be properly reordered in the DOM without unmounting.
+
+The return value of `createFragment` should be treated as an opaque object; you can use the `React.Children` helpers to loop through a fragment but should not access it directly. Note also that we're relying on the JavaScript engine preserving object enumeration order here, which is not guaranteed by the spec but is implemented by all major browsers and VMs for objects with non-numeric keys.
+
+> **Note:**
+>
+> In the future, this may use an API such as
+>
+> ```js
+> return (
+>   <div>
+>     <x:frag key="right">{this.props.rightChildren}</x:frag>,
+>     <x:frag key="left">{this.props.leftChildren}</x:frag>
+>   </div>
+> );
+> ```
+>
+> allowing you to assign keys directly in JSX without adding wrapper elements.
```

Comment:
This last bit is a bit confusing. Perhaps it's just the "this may use an API such as" after explaining that createFragment is the API.